cwms-data-api/src/main/java/cwms/cda/api/project/ProjectChildrenHandler.java

```diff
@@ -0,0 +1,115 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.api.project;
+
+import static com.codahale.metrics.MetricRegistry.name;
+import static cwms.cda.api.Controllers.GET_ALL;
+import static cwms.cda.api.Controllers.LOCATION_KIND_LIKE;
+import static cwms.cda.api.Controllers.OFFICE;
+import static cwms.cda.api.Controllers.PROJECT_LIKE;
+import static cwms.cda.api.Controllers.STATUS_200;
+import static cwms.cda.api.Controllers.requiredParam;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.api.Controllers;
+import cwms.cda.data.dao.JooqDao;
+import cwms.cda.data.dao.project.ProjectChildrenDao;
+import cwms.cda.data.dto.project.ProjectChildren;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+
+
+public class ProjectChildrenHandler implements Handler {
+    public static final String TAGS = "Projects";
+    public static final String PATH = "/projects/children";
+    private final MetricRegistry metrics;
+    private final Histogram requestResultSize;
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    public ProjectChildrenHandler(MetricRegistry metrics) {
+        this.metrics = metrics;
+        requestResultSize = this.metrics.histogram((name(ProjectChildrenHandler.class, Controllers.RESULTS,
+                Controllers.SIZE)));
+    }
+
+
+    @OpenApi(
+            description = "Get a list of project child locations",
+            queryParams = {
+                @OpenApiParam(name = OFFICE, required = true),
+                @OpenApiParam(name = PROJECT_LIKE,  description = "Posix <a href=\"regexp.html\">regular expression</a> matching "
+                        + "against the project_id."),
+                @OpenApiParam(name = LOCATION_KIND_LIKE, description = "Posix <a "
+                        + "href=\"regexp.html\">regular expression</a> matching against the "
+                        + "location kind.  The pattern will be matched against "
+                        + "the valid location-kinds for Project Children:"
+                        + "{\"EMBANKMENT\", \"TURBINE\", \"OUTLET\", \"LOCK\", \"GATE\"}. "
```

Comment:
I think this will also match PUMP

Reply:
I take this back, the relationship between a pump and a project exists only at the water supply contract level. There is no FK from a pump to a Project, it is to a Stream Location.